### PR TITLE
feat(prejoin): unsafe room name warning

### PR DESCRIPTION
**Pull request by gabiborlea**
can be tested at: https://cdumitru.jitsi.net
Simulator Screenshot - iPhone 14 - 2023-05-08 at 17 02 29

### CODE REVIEW

**Review by saghul — APPROVED** [review]
LGTM! LEt's get a second opinion on the copy. I think in the pre-join screen we shouldn't show the last sentence since it's not actionable.

**Comment by gabiborlea** [comment]
Jenkins test this please

File: lang/main.json [hunk 1]
```diff
@@ -974,7 +976,13 @@
     "security": {
         "about": "You can add a $t(lockRoomPassword) to your meeting. Participants will need to provide the $t(lockRoomPassword) before they are allowed to join the meeting.",
         "aboutReadOnly": "Moderator participants can add a $t(lockRoomPassword) to the meeting. Participants will need to provide the $t(lockRoomPassword) before they are allowed to join the meeting.",
-        "insecureRoomNameWarning": "The room name is unsafe. Unwanted participants may join your conference. Consider securing your meeting using the security button.",
+        "insecureRoomNameWarningWeb": "The room name is unsafe. Unwanted participants may join your conference. {{recommendAction}} Learn more about securing you meeting <a href=\"{{securityUrl}}\" rel=\"security\" target=\"_blank\">here</a>.",
```
Comment (saghul) [inline lang/main.json:979]: Replace "conference" with "meeting" here please.

File: lang/main.json [hunk 2]
```diff
@@ -974,7 +976,13 @@
...
+        "insecureRoomNameWarningNative": "The room name is unsafe. Unwanted participants may join your conference. {{recommendAction}} Learn more about securing you meeting {{securityUrl}}",
```
Comment (saghul) [inline lang/main.json:980]: Ditto.

File: react/features/base/util/getUnsafeRoomText.native.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Gets the unsafe room text for the given context.
+ *
+ * @param {Function} t - The translation function.
+ * @param {'meeting'|'prejoin'|'welcome'} context - The given context of the warining.
+ * @returns {Text}
+ */
+export default function getUnsafeRoomText(t: Function, context: 'meeting' | 'prejoin' | 'welcome') {
+    const securityUrl = 'https://jitsi.org/security/';
```
Comment (saghul) [inline react/features/base/util/getUnsafeRoomText.native.ts:15]: Get this from config please.

File: react/features/base/util/getUnsafeRoomText.web.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Gets the unsafe room text for the given context.
+ *
+ * @param {Function} t - The translation function.
+ * @param {'meeting'|'prejoin'|'welcome'} context - The given context of the warining.
+ * @returns {string}
+ */
+export default function getUnsafeRoomText(t: Function, context: 'meeting' | 'prejoin' | 'welcome') {
+    const defaultSecurityUrl = 'https://jitsi.org/security/';
+    const securityUrl = typeof APP === 'undefined'
```
Comment (saghul) [inline react/features/base/util/getUnsafeRoomText.web.ts:12]: You don't need to do this if you are in a .web file, APP is guaranteed to exist.

File: react/features/prejoin/components/native/styles.ts [hunk 5]
```diff
@@ -97,7 +98,54 @@ export const preJoinStyles = {
...
+        maxWidth: 243,
         zIndex: 1
+    },
+    unsafeRoomWarningContainer: {
+        height: '100%',
+        width: '100%',
+        backgroundColor: BaseTheme.palette.ui01,
+        display: 'flex',
+
+        // flexDirection: 'row',
```
Comment (saghul) [inline react/features/prejoin/components/native/styles.ts:110]: ?

**Comment by Calinteodor** [comment]
image

I suggest we choose another text color for the warning description. Maybe keep the warning icon as it is and change the text to white, as it is on prejoin.

**Comment by gabiborlea** [comment]
> image
> 
> I suggest we choose another text color for the warning description. Maybe keep the warning icon as it is and change the text to white, as it is on prejoin. The HERE link text is pretty low visible with the yellow text.

@saghul ??

**Comment by saghul** [comment]
Sounds good, let's make the text white.

**Comment by gabiborlea** [comment]
Jenkins test this please

**Review by saghul — APPROVED** [review]

**Pull request merged**